src/shared/codelens/goCodeLensProvider.ts

```diff
@@ -0,0 +1,195 @@
+
+        const funcSig: string = stripNewLinesAndComments(funcWithBody.substring(sigStart, bodyStart))
+
+        const funcSigParts: string[] = funcSig.split(/\)/)
+        const argTypes: string[] = parseTypes(funcSigParts[0])
+        const retTypes: string[] = funcSigParts.length > 1 ? parseTypes(funcSigParts[1]) : []
+
+        if (argTypes.length > 2 || retTypes.length > 2) {
+            return false
+        }
+
+        // TODO: actually check the types to make sure they are valid
+
+        return true
+    }
+
+    return false
+}
+
+/**
+ * Go allows function signatures to be multi-line, so we should parse these into something more usable.
+ *
+ * @param text String to parse
+ *
+ * @returns Final output without any new lines or comments
+ */
+function stripNewLinesAndComments(text: string): string {
+    const blockCommentRegExp = /\/\*[.*?]\*\//
+    let result: string = ''
+
+    text.split(/\r|\n/).map(s => {
+        const commentStart: number = s.search(/\/\//)
+        s = s.replace(blockCommentRegExp, '')
+        result += commentStart === -1 ? s : s.substring(0, commentStart)
+    })
+
+    return result
+}
+
+/**
+ * Finds all types of the parameter list, stripping out names if they exist.
+ * Parenthesis are stripped from the list if they exist.
+ *
+ * @param params List of parameters delimeted by commas
+ *
+ * @returns A list of types mapping 1:1 with the input
+ */
+function parseTypes(params: string): string[] {
+    params = params.replace(/\(|\)/, '')
+    const types: string[] = []
+
+    let lastType: number = 0
+    const paramParts = params.split(',')
+
+    // Names of parameters must either be all present or all absent: https://golang.org/ref/spec#Function_types
+    for (let i = 0; i < paramParts.length; i++) {
+        const parts: string[] = paramParts[i].trim().split(/\s+/)
+        const type: string = parts.length > 1 ? parts[1] : parts[0]
+
+        // Go allows types to be assigned to multiple parameters, e.g. (x, y, z int) === (x int, y int, z int)
+        if (parts.length > 1) {
+            for (let j = lastType; j < i; j++) {
+                types[j] = type
+            }
+            lastType = i + 1
+        }
+
+        types.push(type)
+    }
+
+    return types
+}
+
+/**
+ * Checks if the Go extension exists and is active.
+ */
+async function checkForGoExtension(): Promise<boolean> {
+    const extension = vscode.extensions.getExtension(VSCODE_EXTENSION_ID.go)
+
+    if (extension) {
+        if (extension.isActive) {
+            return true
+        }
+
+        getLogger().info('Go CodeLens provider is activating the Go extension...')
+
+        try {
+            await extension.activate()
+            getLogger().info('Go extension activated!')
+
+            return true
+        } catch (err) {
+            getLogger().info('Failed to activate Go extension. The toolkit will have reduced functionality.')
+            getLogger().debug('Extension activation failed: %O', err as Error)
```

Comment:
- Let's avoid log noise as much as possible. E.g. here we can show one or the other, not both.
- Also let's be precise where we can, mention the extension id or even a URL to its marketplace page. "The Go extension" means nothing to anyone except us at this particular moment in time :) There could be many different Go extensions.
- Avoid entropy: use the same message start for DEBUG and INFO

```suggestion
            if (getLogger().logLevelEnabled('debug')) {
              getLogger().debug('Failed to activate extension "%s": %O', VSCODE_EXTENSION_ID.go , err as Error)
            } else {
              getLogger().info('Failed to activate extension "%s". AWS codelenses will not appear in .go files.')
            }
```